src/tests/ftest/launch.py

```diff
@@ -20,164 +20,552 @@
+        test_yaml (str): test yaml file
+
+    Returns:
+        list: a unique list of hosts specified in the test's yaml file
+
+    """
+    yaml_data = get_yaml_data(test_yaml)
+    server_keys = [
+        "test_machines",
+        "test_servers",
+        "daos_servers",
+        "servers",
+        "test_machines1",
+        "test_machines2",
+        "test_machines2a",
+        "test_machines3",
+        "test_machines6"]
+    client_keys = [
+        "test_clients",
+        "clients",
+    ]
+    host_set = set()
+    found_client_key = False
+    matches = find_values(yaml_data, server_keys + client_keys)
+    for key, value in matches.items():
+        host_set.update(value)
+        if key in client_keys:
+            found_client_key = True
+
+    # Include this host as a client if no clients are specified
+    if not found_client_key:
+        host_set.add(socket.gethostname().split(".")[0])
+
+    return sorted(list(host_set))
+
+
+def clean_logs(test_yaml):
+    """Remove the test log files on each test host.
+
+    Args:
+        test_yaml (str): yaml file containing host names
+    """
+    # Use the default server yaml and then the test yaml to update the default
+    # DAOS log file locations.  This should simulate how the test defines which
+    # log files it will use when it is run.
+    log_files = get_log_files(test_yaml, get_log_files(BASE_LOG_FILE_YAML))
+    host_list = get_hosts_from_yaml(test_yaml)
+    command = "ssh {{host}} \"rm -fr {}\"".format(" ".join(log_files.values()))
+    print("Cleaning logs on {}".format(host_list))
+    spawn_commands(host_list, command)
+
+
+def archive_logs(avocado_logs_dir, test_yaml):
+    """Copy all of the host test log files to the avocado results directory.
+
+    Args:
+        avocado_logs_dir ([type]): [description]
+        test_yaml (str): yaml file containing host names
+    """
+    this_host = socket.gethostname().split(".")[0]
+    log_files = get_log_files(TEST_LOG_FILE_YAML)
+    host_list = get_hosts_from_yaml(test_yaml)
+    doas_logs_dir = os.path.join(avocado_logs_dir, "latest", "daos_logs")
+
+    # Create a subdirectory in the avocado logs directory for this test
+    print("Archiving host logs from {} in {}".format(host_list, doas_logs_dir))
+    get_output("mkdir {}".format(doas_logs_dir))
+
+    # Create a list of log files that are not directories
+    non_dir_files = [
+        log_file for log_file in log_files.values()
+        if os.path.splitext(os.path.basename(log_file))[1] != ""]
+
+    # Copy any log files that exist on the test hosts and remove them from the
+    # test host if the copy is successful
+    command = "ssh {{host}} 'set -e; {}'".format(
+        "for file in {}; do if [ -e $file ]; then "
+        "scp $file {}:{}/${{{{file##*/}}}}-{{host}} && rm -fr $file; fi; "
+        "done".format(" ".join(non_dir_files), this_host, doas_logs_dir))
+    spawn_commands(host_list, command)
+
+
+def rename_logs(avocado_logs_dir, test_file):
```

Comment:
We have same test file under different folder for example pool/attribute.py container/attribute.py. When we rename the log it will be same folder renamed so not sure if this will be problem in Ci or not.

Reply:
Yeah.  Now that he points it out I think @ravalsam is correct.

I had an implementation WIP of this sort of thing and I was using both the dir and test name for this reason.  Sorry I missed this on review.

Reply:
Agreed.  I will include this change in https://github.com/daos-stack/daos/pull/496

Reply:
@phender: I don't see this in #496, but that is good.  That PR should just address it's stated goal.

So to resolve @ravalsam's point, perhaps we should go forward with the `--job-category` solution, either by monkey-patching that into our avocado, or by writing our own code to emulating what it will end up doing?  I tend to lean towards monkey-patching so that we are writing the code for the proper avocado feature.

An example of similar monkey-patching can be seen here:
https://github.com/daos-stack/daos/blob/c768f0c1656b1b90f60fb4139a8f591fbe12beb8/ftest.sh#L254-L312

You will need to massage the upstream patch to have the fully qualified pathname such as in lines 278-279 above.

I do also wonder though, if we are getting to a point where we should keep the patches in files, perhaps in `src/test/ftest/avocado-patches/` (or something like that) rather than inline in ftest.sh.